TestingLowerBounds/ForMathlib/ERealLogExp.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2024 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import TestingLowerBounds.ForMathlib.EReal
+import Mathlib.Analysis.SpecialFunctions.Log.Basic
+
+/-!
+
+# Logarithm and exponential in the extended reals
+
+## Main definitions
+
+* `EReal.log : ℝ≥0∞ → EReal`
+* `EReal.exp : EReal → ℝ≥0∞`
+
+## Main statements
+
+* `EReal.log_exp : EReal.log (EReal.exp x) = x`
+* `EReal.exp_log : EReal.exp (EReal.log x) = x`
+
+-/
+
+open scoped ENNReal
+
+namespace EReal
+
+section Log
+
+/-- Logarithm as a function from `ℝ≥0∞` to `EReal`. -/
+noncomputable
+def log : ℝ≥0∞ → EReal
+| ∞ => ⊤
+| x => if x = 0 then ⊥ else Real.log x.toReal
+
+@[simp] lemma log_zero : log 0 = ⊥ := by simp [log]
+@[simp] lemma log_top : log ∞ = ⊤ := by simp [log]
+
```

Comment:
```suggestion
@[simp] lemma log_zero : log 0 = ⊥ := if_pos rfl
@[simp] lemma log_top : log ∞ = ⊤ := rfl
```

Reply:
I'll change it! That suggestion is not at the right line, so I cannot apply it through the github interface though.

Reply:
Ok thanks, I'm trying to do the review directly from VScode, so I can run the code, but I still have to figure out exactly how the comments behave.